server/data/data-ldap/src/main/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepository.java

```diff
@@ -770,7 +770,7 @@ public String getUser(MailAddress mailAddress) throws UsersRepositoryException {
         if (supportVirtualHosting()) {
             return mailAddress.asString();
         } else {
-            return mailAddress.getLocalPart();
+            return mailAddress.getLocalPart().toLowerCase();
```

Comment:
`toLowerCase()` needs to take a Locale as parameter (as the locale might impact the operation.

Here you might want: `toLowerCase(Locale.US)`

Reply:
Additionally, we need tests about these changes.

Have a look at `AbstractUsersRepositoryTest`